05.wc/wc.rb

```diff
@@ -0,0 +1,70 @@
+# ! /usr/bin/env ruby
+# frozen_string_literal: true
+
+require 'optparse'
+
+STDIN_FILENAME = '-'
+TOTAL = '合計'
+
+def main
+  options = parse_options(ARGV)
+  rows = []
+  ARGF.each(nil) do |input_text|
+    counts = {}
+
+    counts[:filename] = ARGF.filename
+    counts[:line] = input_text.lines.count if options[:l]
+    counts[:word] = input_text.split(/\s+/).size if options[:w]
+    counts[:byte] = input_text.size if options[:c]
+    rows << counts
+  end
+  rows << calculate_total_rows(rows) if rows.size > 1
+  max_widths = calculate_max_widths(rows)
+  show_rows(rows, max_widths)
```

Comment:
```suggestion
  show_rows(rows)
```

rowsだけ渡してもらえればあとはいい感じに出力しておきまっせ、という設計の方がメソッドの再利用性が高そうです